hmiddleware/basicauth/grpc.go

```diff
@@ -34,6 +35,22 @@ func GRPCAuthFunc(checker *Checker) func(ctx context.Context) (context.Context,
 			return nil, grpc.Errorf(codes.PermissionDenied, "permission denied") //nolint:staticcheck
 		}
 
+		if checker.roleValidator != nil {
+			// According to grpc docs, this method will be in the form:
+			// "/service/method".
+			method, ok := grpc.Method(ctx)
+			if !ok {
+				return nil, grpc.Errorf(codes.PermissionDenied, "permission denied. no grpc method found on context") //nolint:staticcheck
```

Comment:
Is the error message relayed back to the client?

Reply:
Mmm good question. Don't know, but I can check what we do